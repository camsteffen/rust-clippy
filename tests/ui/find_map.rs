#![warn(clippy::all, clippy::pedantic)]

#[derive(Debug, Copy, Clone)]
enum Flavor {
    Chocolate,
}

#[derive(Debug, Copy, Clone)]
enum Dessert {
    Banana,
    Pudding,
    Cake(Flavor),
}

fn main() {
    let desserts_of_the_week = vec![Dessert::Banana, Dessert::Cake(Flavor::Chocolate), Dessert::Pudding];

    let a = ["lol", "NaN", "2", "5", "Xunda"];

    let _: Option<i32> = a.iter().find(|s| s.parse::<i32>().is_ok()).map(|s| s.parse().unwrap());
}

fn no_lint() {
    let _ = vec![1].into_iter().find(|n| *n > 5).map(|n| n + 7);
}
