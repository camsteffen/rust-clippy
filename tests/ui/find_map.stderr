error: called `find(..).map(..)` on an `Iterator`
  --> $DIR/find_map.rs:20:26
   |
LL |     let _: Option<i32> = a.iter().find(|s| s.parse::<i32>().is_ok()).map(|s| s.parse().unwrap());
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::find-map` implied by `-D warnings`
   = help: this is more succinctly expressed by calling `.find_map(..)` instead

error: aborting due to previous error

